community/kernel/src/main/java/org/neo4j/graphdb/GraphDatabaseService.java

```diff
@@ -183,10 +187,55 @@
      * space).
      *
      * @return all relationship types in the underlying store
-     * @deprecated this operation can be found in {@link GlobalGraphOperations} instead.
      */
-    @Deprecated
-    Iterable<RelationshipType> getRelationshipTypes();
+    ResourceIterable<RelationshipType> getRelationshipTypes();
+
+    /**
+     * Returns all relationship types currently in use in the underlying store. Relationship types are
+     * added to the underlying store the first time they are used in a successfully committed
+     * {@link Node#createRelationshipTo node.createRelationshipTo(...)}. Note that this method is
+     * guaranteed to return all known relationship types, and it guarantees that it won't return
+     * "historic" relationship types that no longer have any relationships in the graph.
+     *
+     * @return all relationship types in use in the underlying store
+     */
+    ResourceIterable<RelationshipType> getAllRelationshipTypesInUse();
+
+    /**
+     * Returns all labels currently in the underlying store. Labels are added to the store the first time
+     * they are used. This method guarantees that it will return all labels currently in use. However,
+     * it may also return <i>more</i> than that (e.g. it can return "historic" labels that are no longer used).
+     *
+     * Please take care that the returned {@link ResourceIterable} is closed correctly and as soon as possible
+     * inside your transaction to avoid potential blocking of write operations.
+     *
+     * @return all labels in the underlying store.
+     */
+    ResourceIterable<Label> getAllLabels();
+
+    /**
+     * Returns all labels currently in use in the underlying store. Labels are added to the store the first time
+     * they are used. This method guarantees that it will return all labels currently in use by filtering out
+     * "historic" labels that are no longer used.
+     *
+     * Please take care that the returned {@link ResourceIterable} is closed correctly and as soon as possible
+     * inside your transaction to avoid potential blocking of write operations.
+     *
+     * @return all labels in use in the underlying store.
+     */
+    ResourceIterable<Label> getAllLabelsInUse();
```

Comment:
Yes, I agree that having this consistent with the methods for relationship types would be useful.